rich/traceback.py

```diff
@@ -369,20 +440,21 @@ def safe_str(_object: Any) -> str:
                         filename = os.path.join(_IMPORT_CWD, filename)
                 if frame_summary.f_locals.get("_rich_traceback_omit", False):
                     continue
+
+                if show_locals:
+                    local_variables = {
+                        k: mask_local_variable(key=k, value=v, patterns=mask_locals)
```

Comment:
Excluding `pretty.traverse` will break the advanced formatting of data structures, and a number of other Rich features. I think the solution has to involve `pretty.transverse`.